Common/NearestNeighbour/src/main/java/org/tribuo/common/nearest/KNNModel.java

```diff
@@ -161,43 +166,24 @@
         } else {
             List<Prediction<T>> predictions = new ArrayList<>();
             List<Prediction<T>> innerPredictions = new ArrayList<>();
-            PriorityQueue<OutputDoublePair<T>> queue = new PriorityQueue<>(k, (a,b) -> Double.compare(b.value, a.value));
-            BiFunction<SparseVector,SparseVector,Double> distanceFunc;
-            switch (distance) {
-                case L1:
-                    distanceFunc = (a,b) -> b.l1Distance(a);
-                    break;
-                case L2:
-                    distanceFunc = (a,b) -> b.l2Distance(a);
-                    break;
-                case COSINE:
-                    distanceFunc = (a,b) -> b.cosineDistance(a);
-                    break;
-                default:
-                    throw new IllegalStateException("Unknown distance function " + distance);
-            }
+
+            NeighboursBruteForceFactory nbff = new NeighboursBruteForceFactory(distType, numThreads);
```

Comment:
Maybe this should become a field on this object? We can fabricate one during deserialization for old KNNs.

Reply:
I agree that this should be a field. I was thinking it would be better to add these changes when there is more than one `NeighboursQueryFactory` implementation to which it could be assigned. I'm also fine to add these changes now, since the k-d tree should be coming soon. Let me know.

Reply:
Later is fine.